crux-core/src/crux/api/v2/CruxNode.java

```diff
@@ -0,0 +1,75 @@
+package crux.api.v2;
+
+import clojure.lang.Keyword;
+import clojure.lang.PersistentVector;
+import crux.api.Crux;
+import crux.api.ICruxAPI;
+import crux.api.IndexVersionOutOfSyncException;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static crux.api.v2.Database.database;
+import static crux.api.v2.TxResult.txResult;
+
+public class CruxNode {
+    private final ICruxAPI node;
+
+    private CruxNode(Topology topology) throws IndexVersionOutOfSyncException {
+        this.node = Crux.startNode(topology.toEdn());
+    }
+
+    public static CruxNode cruxNode(Topology topology) throws IndexVersionOutOfSyncException {
+        return new CruxNode(topology);
+    }
+
+    @SuppressWarnings("unchecked")
+    public TxResult submitTx(Iterable<Operation> ops) {
+        PersistentVector txVector = PersistentVector.create();
+        for(Operation op: ops) {
+            txVector = txVector.cons(op.toEdn());
+        }
+
+        Map<Keyword,Object> result = node.submitTx(txVector);
+        Date txTime = (Date) result.get(Keyword.intern("crux.tx/tx-time"));
+        Long txId = (Long) result.get(Keyword.intern("crux.tx/tx-id"));
+        return txResult(txTime, txId);
+    }
+
+    @SuppressWarnings("unchecked")
+    public TxResult submitTx(Operation... ops) { return submitTx(Arrays.asList(ops)); }
+
+    public Database db() {
+        return database(node);
+    }
+
+    public Database db(Date validTime) {
+        return database(node, validTime);
+    }
+
+    public Database db(Date validTime, Date transactionTime) {
+        return database(node, validTime, transactionTime);
+    }
+
+    public Document document(Object contentHash) {
```

Comment:
It's not clear now, whether `CruxId` is acceptable here or not, or how to obtain an instance of `contentHash` from the public v2 API. If this is an internal method for Clojure query engine, it might be less useful in this Java-wrapper.

The `Document` itself does not offer `contentHash`, only the `id`. Maybe a new proposed `DocumentSnapshot` should hide the concept of content hashes from the public API.

Reply:
agreed, this is unclear. have made this method private until we can clarify it